extensions/sql-migration/src/dialog/skuRecommendationResults/GenerateProvisioningScriptDialog.ts

```diff
@@ -100,17 +100,18 @@ export class GenerateProvisioningScriptDialog {
 		saveTemplateButton.onDidClick(async () => {
 			const folder = await utils.promptUserForFolder();
 			if (folder) {
-				if (this._armTemplateText) {
-					let templateName = utils.generateTemplatePath(this.model, this._targetType);
-					let destinationFilePath = path.join(folder, templateName);
-					try {
-						fs.writeFileSync(destinationFilePath!, this._armTemplateText);
-						void vscode.window.showInformationMessage(constants.SAVE_TEMPLATE_SUCCESS);
-					}
-					catch (e) {
-						logError(TelemetryViews.ProvisioningScriptWizard, 'ArmTemplateSavetoLocalError', e);
-						void vscode.window.showErrorMessage(constants.SAVE_TEMPLATE_FAIL);
+				const templates = this.model._armTemplateResult.templates!;
+				try {
+					for (let i = 0; i < templates.length; i++) {
```

Comment:
templates.length will break if templates is undefined. Since templates type is string|undefined. Instead you can use templates?.length